### ISSUE TITLE
Money field and TextColunm->money() bug

### ISSUE THREAD

**Issue by HomaEEE** [filamentphp/filament#3098]
### Package

filament/filament

### Package Version

v2.13

### Laravel Version

v9

### Livewire Version

v2

### PHP Version

PHP 8.1

### Bug description

Mysql field -  `$table->decimal('price', 10, 2)->nullable(); ` 
From field -  `TextInput::make('price')->mask(fn (TextInput\Mask $mask) => $mask->money('$', ',', 2))`
Table field -  `TextColumn::make('price')->money('uah')->label('Цена')` 

In database I store value - 7.89 (its right)
In Table - $ 0.08 (its error for display)

<img width="879" alt="Снимок экрана 2022-07-13 в 00 26 52" src="https://user-images.githubusercontent.com/1429343/178598392-cca3f14d-7596-4d1a-9c77-e64cd2b0eb61.png">

But in form - $ 7.89  (its right) 

<img width="971" alt="Снимок экрана 2022-07-13 в 00 25 05" src="https://user-images.githubusercontent.com/1429343/178598344-52c9a228-ebdb-49d6-8bfc-75538198403d.png">

**Comment by simonbuehler** [comment]
you would need ` TextColumn::make('price')->money('uah', true)->label('Цена')` to convert from your decimal 7.9 calue (  = ~8 cents ) as it by default gets constructed from minor (https://github.com/akaunting/laravel-money#usage)

in the form no actual `Money` instance is created, ist just a mask

can be closed  imo

**Comment by blite** [comment]
This is actually not correct in the docs.  The TextColumn encourages you to store as integer (which is correct).  The examples on https://filamentphp.com/docs/2.x/forms/fields#input-masking

```
TextInput::make('cost')->mask(fn (TextInput\Mask $mask) => $mask->money('$', ',', 2))
```

IMO the examples should be congruent and work together or at least link to instructions for https://github.com/akaunting/laravel-money#usage that would make them so.

**Comment by simonbuehler** [comment]
i agree that for many standard usecases a integer value of the minor unit ( e.g. cents) is ok, but when handling stocks and exchange rate calculations or when handling multiple different currencies a higher decimal scale is needed than the scale the main currency provides.
just for reference: https://github.com/brick/money is also a great library for handling currencies

**Comment by blite** [comment]
The issue I am having and that I believe op is reporting is that the 
```
TextInput::make('price')->mask(fn (TextInput\Mask $mask) => $mask->money('$', ',', 2))
```
only works with decimals

while
```
TextColumn::make('price')->money('uah')->label('Цена')
```
works with integers.  

As far as I can see the imask.js library doesn't support integer money.  Which honestly means it either needs to get it added or we should seek a different library.  The fact that I can't even find an issue about this on https://github.com/uNmAnNeR/imaskjs is somewhat worrying.  

the demo app uses  decimals https://github.com/filamentphp/demo/blob/main/database/migrations/2021_12_13_164519_create_shop_products_table.php#L27
which needs to be fixed.

Op likely copied the decimals from the demo app and doesn't understand that https://github.com/akaunting/laravel-money#usage  (that column->money() uses is correctly expecting integers).  The shortcut is probably to use https://github.com/akaunting/laravel-money#usage 's convert functionality, but it would still be not the best example to be giving everyone.

**Comment by simonbuehler** [comment]
imask.js has multiple issues and there are plans to replace it [_soon_ ](https://github.com/filamentphp/filament/issues/1624#issuecomment-1092637738)™

**Comment by blite** [comment]
anyway I added the documentation pull request https://github.com/filamentphp/filament/pull/3190 which for now at least makes them in sync even if not ideal.
